test/e2e/performanceprofile/functests/utils/jira/status.go

```diff
@@ -26,28 +33,72 @@ type JiraIssueStatusResponse = struct {
 func RetrieveJiraStatus(key string) (*JiraIssueStatusResponse, error) {
 	local_params := url.Values{}
 	local_params.Set("fields", "summary,status")
-
 	fullUrl := JIRA_BASE_URL + "/rest/api/2/issue/" + key + "?" + local_params.Encode()
-	req, err := http.NewRequest("GET", fullUrl, nil)
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to create request for Jira status of %s", key)
-	}
-	ret, err := http.DefaultClient.Do(req)
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to create client to get jira status of %s", key)
+	jiraResponse := JiraIssueStatusResponse{}
+	for retryCount := 0; retryCount <= MAX_RETRIES; retryCount++ {
+		req, err := http.NewRequest("GET", fullUrl, nil)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to create request for Jira status of %s", key)
+		}
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to create client to get jira status of %s", key)
+		}
+		defer resp.Body.Close()
+		switch resp.StatusCode {
+		case http.StatusOK:
+			decoder := json.NewDecoder(resp.Body)
+			err = decoder.Decode(&jiraResponse)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to decode jira status of %s", key)
+			}
+		case http.StatusTooManyRequests:
+			if retryAfter := resp.Header.Get("Retry-After"); retryAfter != "" {
+				retryTime, err := convertRetryAfterTime(retryAfter)
+				if err != nil {
+					return nil, err
```

Comment:
Wrong Retry-After header could just log the error and fallback to exponential backoff. Unless we want to know the parsing broke right away.

Reply:
fixed in latest commit